cli/tools/lint/mod.rs

```diff
@@ -191,7 +175,7 @@ async fn lint_files(
   let reporter_lock =
     Arc::new(Mutex::new(create_reporter(reporter_kind.clone())));
   let has_error = Arc::new(AtomicFlag::default());
-
+  let linter = create_linter(lint_rules.clone());
```

Comment:
This might works? creating an instance outside the future and cloning it when it is used inside the future.